core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -93,13 +93,26 @@ protected void killComputer(Computer c) {
     * Package-protected, but accessed API
     * ============================================================================================================== */
 
-    /*package*/ final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<>();
+    private Set<String> disabledAdministrativeMonitors = new HashSet<>();
 
-    @Restricted(NoExternalUse.class)
-    public CopyOnWriteArraySet<String> getDisabledAdministrativeMonitors(){
+    /**
+     * Get the disabled administrative monitors
+     *
+     * @since TODO
+     */
+    public Set<String> getDisabledAdministrativeMonitors(){
     	return disabledAdministrativeMonitors;
     }
-    
+
+    /**
+     * Set the disabled administrative monitors
+     *
+     * @since TODO
+     */
+    public void setDisabledAdministrativeMonitors(Set<String> disabledAdministrativeMonitors){
+        this.disabledAdministrativeMonitors = disabledAdministrativeMonitors;
```

Comment:
Probably should be cloned here instead of taking a reference to an external object

Reply:
Why’s that? It’s just a setter

Reply:
both stapler and jcasc will handle creating a clone that should be safe for Jenkins core to use.

Reply:
Spotbugs normally warns against taking a reference to an outside object

Reply:
not on collections or simple object types.

This would be the case on date objects.